### PR TITLE
Db/integration tests

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

[CLI integration tests](https://github.com/OpsLevel/team-platform/issues/432)

## Changelog

Add integration tests on CRUD operations in cli. Creates, Updates, Deletes actual test resources

- [X] List your changes here
- [ ] Make a `changie` entry, N/A testing only

## Tophatting

`task test-integration`

### CODE REVIEW

File: Taskfile.yml
Comment (taimoorgit) [inline Taskfile.yml:55]: I noticed that these 2 are the same, could add `test` as an alias for this task. https://taskfile.dev/usage/#task-aliases

File: src/cmd/util_test.go
Comment (taimoorgit) [inline src/cmd/util_test.go:49]: Function can be changed to return `([]byte, error)` so that we can handle the error on line 53 and also on line 36 since `execCmd` calls this function.

File: src/cmd/util_test.go
Comment (taimoorgit) [inline src/cmd/util_test.go:55]: Remove this commented out line, this call is done on line 60

File: src/cmd/util_test.go
Comment (taimoorgit) [inline src/cmd/util_test.go:64]: Also recommend that this function returns `(*T, error)` so that it can be handled by the calling test for example user_test.go line 37.

File: src/cmd/util_test.go
Comment (taimoorgit) [inline src/cmd/util_test.go:73]: Var should not be named `userFileName`, conflicts with the name of another package scoped variable.

File: src/cmd/util_test.go
Comment (taimoorgit) [inline src/cmd/util_test.go:41]: Recommend that this change to returning `(*os.File, *os.File, error)` so that the error from `os.Pipe()` is handled.

File: src/cmd/team_test.go
Comment (taimoorgit) [inline src/cmd/team_test.go:14]: This would be much more readable if divided into functions like `Test_TeamCreate`, `Test_TeamUpdate` and so on, I would be able to quickly see what team related functions have tests by glancing at the functions defined in this file.

File: src/cmd/user_test.go
Comment (taimoorgit) [inline src/cmd/user_test.go:16]: Same here (separate test functions)

File: src/cmd/util_test.go
Comment (taimoorgit) [inline src/cmd/util_test.go:26]: Could we do this? 

Rename `execCmd` to `ExecCmd` and keep it in the `cmd` package (since it needs access to `rootCmd`), and then remove everything else added in this PR a new `cmd_test` package?

This would give us better separation between CLI code and tests. However the CLI is not really a "library" we are providing so I don't hold this opinion strongly.

**Review by taimoorgit — CHANGES_REQUESTED** [review]
Had some feedback,

1. We should handle all errors instead of ignoring them with `_`. If the error is really unusual for a test and it is tedious to have the calling function handle them, can we just `panic()` if `err != nil`?

2. Tests would be much more easy to read if each object had multiple `Test_` functions rather than only `Test_...CRUD` which does everything. 

3. We could add a `cmd_test` package (I don't hold this opinion strongly)

File: Taskfile.yml
Comment (davidbloss) [reply Taskfile.yml:55]: The difference is `task test` runs tests in `./src/common/...` and `task test-integration` runs tests in `./src/cmd/...`

These should be separated as `task test` locally tests code and is part of `task ci`, whereas `task test-integration` is intended for pre-deploy integration testing and is intentionally kept out of regular testing

File: src/cmd/team_test.go
Comment (davidbloss) [reply src/cmd/team_test.go:14]: The code could be divided up into smaller functions but we can't run something like `Test_TeamUpdate` in isolation. We first need to create some test team to update

File: src/cmd/team_test.go
Comment (taimoorgit) [inline src/cmd/team_test.go:24]: Request that we replace these `t.Error(err)` and `t.FailNow()` calls with `t.Fatal(err)`

https://pkg.go.dev/testing#T.FailNow

>Error is equivalent to Log followed by Fail.

>Fatal is equivalent to Log followed by FailNow.

Therefore `t.Fatal(err)` is the better call to make.

File: src/cmd/team_test.go
Comment (taimoorgit) [inline src/cmd/team_test.go:29]: When it comes to all of these CRUD functions that the code has been divided into, we could actually pass in `t *testing.T` into them, and have those functions call `t.Fatal(err)` so that we don't have to examine return values and this function could be much shorter.

**Review by taimoorgit — CHANGES_REQUESTED** [review]
Great revision, I have another recommendation to consider before merging

File: src/cmd/team_test.go
Comment (davidbloss) [reply src/cmd/team_test.go:29]: There are problems with automatically failing in each function.

If `updateTeam(...)` fails to update the created test team, we need to the flexibility to delete the team outside of that function. Other tests down the road might make user of `updateTeam()` or even `getTeam()` and those tests would need to determine how to handle those situations. We might want to run `getTeam()` to prove a team has been deleted for instance.

**Review by taimoorgit — APPROVED** [review]

**Pull request merged**
